src/Provider/Pelias/Pelias.php

```diff
@@ -0,0 +1,251 @@
+     *
+     * @throws \Geocoder\Exception\Exception
+     */
+    protected function getGeocodeQueryUrl(GeocodeQuery $query, array $query_data = []): string
+    {
+        $address = $query->getText();
+
+        // This API doesn't handle IPs
+        if (filter_var($address, FILTER_VALIDATE_IP)) {
+            throw new UnsupportedOperation('The Pelias provider does not support IP addresses, only street addresses.');
+        }
+
+        $data = [
+            'text' => $address,
+            'size' => $query->getLimit(),
+        ];
+
+        return sprintf('%s/search?%s', $this->root, http_build_query(array_merge($data, $query_data)));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function geocodeQuery(GeocodeQuery $query): Collection
+    {
+        return $this->executeQuery($this->getGeocodeQueryUrl($query));
+    }
+
+    /**
+     * @param ReverseQuery $query
+     * @param array        $query_data Additional query data (API key for instance).
+     *
+     * @return string
+     *
+     * @throws \Geocoder\Exception\Exception
+     */
+    protected function getReverseQueryUrl(ReverseQuery $query, array $query_data = []): string
+    {
+        $coordinates = $query->getCoordinates();
+        $longitude = $coordinates->getLongitude();
+        $latitude = $coordinates->getLatitude();
+
+        $data = [
+            'point.lat' => $latitude,
+            'point.lon' => $longitude,
+            'size' => $query->getLimit(),
+        ];
+
+        return sprintf('%s/reverse?%s', $this->root, http_build_query(array_merge($data, $query_data)));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reverseQuery(ReverseQuery $query): Collection
+    {
+        return $this->executeQuery($this->getReverseQueryUrl($query));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName(): string
+    {
+        return 'pelias';
+    }
+
+    /**
+     * @param $url
+     *
+     * @return Collection
+     */
+    protected function executeQuery(string $url): AddressCollection
+    {
+        $content = $this->getUrlContents($url);
+        $json = json_decode($content, true);
+
+        if (isset($json['meta'])) {
+            switch ($json['meta']['status_code']) {
+                case 401:
+                case 403:
+                    throw new InvalidCredentials('Invalid or missing api key.');
+                case 429:
+                    throw new QuotaExceeded('Valid request but quota exceeded.');
+            }
+        }
+
+        if (!isset($json['type']) || 'FeatureCollection' !== $json['type'] || !isset($json['features']) || 0 === count($json['features'])) {
+            return new AddressCollection([]);
+        }
+
+        $locations = $json['features'];
+
+        if (empty($locations)) {
+            return new AddressCollection([]);
+        }
+
+        $results = [];
+        foreach ($locations as $location) {
+            $bounds = [
+                'south' => null,
```

Comment:
Should we be initialising this to null?

> Changed: Location::getBounds will return null or a Bounds object with coordinates data. It will never return Bounds without data.

Reply:
Nice catch, thanks!